### ISSUE TITLE
[Bug]: [Nightly]Milvus crashed with plenty of cases failed when load collection

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#22159]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:7c49530
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    kafka
- SDK version(e.g. pymilvus v2.0.0rc2):2.3.0.dev34
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Milvus crashed with plenty of cases failed when load collection.
error message:
```
[2023-02-13T19:28:08.340Z] [2023-02-13 18:45:03 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=failed to load collection, err=no enough nodes to create replicas[NoEnoughNode])>, <Time:{'RPC start': '2023-02-13 18:45:03.258241', 'RPC error': '2023-02-13 18:45:03.262222'}> (decorators.py:108)
[2023-02-13T19:28:08.340Z] [2023-02-13 18:45:03 - ERROR - ci_test]: Traceback (most recent call last):
[2023-02-13T19:28:08.340Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
[2023-02-13T19:28:08.340Z]     res = func(*args, **_kwargs)
[2023-02-13T19:28:08.340Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
[2023-02-13T19:28:08.340Z]     return func(*arg, **kwargs)
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 366, in load
[2023-02-13T19:28:08.340Z]     conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2023-02-13T19:28:08.340Z]     raise e
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2023-02-13T19:28:08.340Z]     return func(*args, **kwargs)
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2023-02-13T19:28:08.340Z]     ret = func(self, *args, **kwargs)
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
[2023-02-13T19:28:08.340Z]     raise e
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2023-02-13T19:28:08.340Z]     return func(self, *args, **kwargs)
[2023-02-13T19:28:08.340Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 661, in load_collection
[2023-02-13T19:28:08.340Z]     raise MilvusException(response.error_code, response.reason)
[2023-02-13T19:28:08.340Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to load collection, err=no enough nodes to create replicas[NoEnoughNode])>
[2023-02-13T19:28:08.340Z]  (api_request.py:39)
[2023-02-13T19:28:08.340Z] [2023-02-13 18:45:03 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=failed to load collection, err=no enough nodes to create replicas[NoEnoughNode])> (api_request.py:40)
```

### Expected Behavior

succeed

### Milvus Log

1. link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/282/pipeline/155/
2. log: [artifacts-milvus-distributed-kafka-nightly-282-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/282/artifact/artifacts-milvus-distributed-kafka-nightly-282-pymilvus-e2e-logs.tar.gz)
3. milvus log: 
```
2023-02-14T02:07:42.628186737+08:00 stderr F panic: runtime error: invalid memory address or nil pointer dereference
2023-02-14T02:07:42.628216937+08:00 stderr F [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x29d4f2b]
2023-02-14T02:07:42.628222151+08:00 stderr F
2023-02-14T02:07:42.628225371+08:00 stderr F goroutine 123 [running]:
2023-02-14T02:07:42.628229706+08:00 stderr F panic({0x335caa0, 0x4e59f00})
2023-02-14T02:07:42.628234088+08:00 stderr F    /usr/local/go/src/runtime/panic.go:941 +0x397 fp=0xc000f82ff8 sp=0xc000f82f38 pc=0x11edcf7
2023-02-14T02:07:42.62823795+08:00 stderr F runtime.panicmem(...)
2023-02-14T02:07:42.628242629+08:00 stderr F    /usr/local/go/src/runtime/panic.go:220
2023-02-14T02:07:42.628246905+08:00 stderr F runtime.sigpanic()
2023-02-14T02:07:42.628251515+08:00 stderr F    /usr/local/go/src/runtime/signal_unix.go:818 +0x336 fp=0xc000f83048 sp=0xc000f82ff8 pc=0x1204d16
2023-02-14T02:07:42.628282846+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*QueryNode).LoadSegments(0xc000ede500, {0x3adbcd8?, 0xc000eb02d0}, 0xc00028f400)
2023-02-14T02:07:42.628302295+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:432 +0x8b fp=0xc000f834e8 sp=0xc000f83048 pc=0x29d4f2b
2023-02-14T02:07:42.628306142+08:00 stderr F github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).LoadSegments(0x0?, {0x3adbcd8?, 0xc000eb02d0?}, 0x0?)
2023-02-14T02:07:42.628309685+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:270 +0x2b fp=0xc000f83518 sp=0xc000f834e8 pc=0x2bbd16b
2023-02-14T02:07:42.628312585+08:00 stderr F github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_LoadSegments_Handler.func1({0x3adbcd8, 0xc000eb02d0}, {0x35ff8e0?, 0xc00028f400})
2023-02-14T02:07:42.628322616+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5613 +0x78 fp=0xc000f83558 sp=0xc000f83518 pc=0x248fc78
2023-02-14T02:07:42.628329969+08:00 stderr F github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor({0x3adbcd8?, 0xc000eb02d0?}, {0x35ff8e0, 0xc00028f400}, 0xc000f835b8?, 0xc000132e40)
2023-02-14T02:07:42.628335285+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22 +0x49 fp=0xc000f83588 sp=0xc000f83558 pc=0x24756e9
2023-02-14T02:07:42.628357707+08:00 stderr F github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3adbcd8?, 0xc000eb02d0?}, {0x35ff8e0?, 0xc00028f400?})
2023-02-14T02:07:42.628370023+08:00 stderr F    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc000f835c8 sp=0xc000f83588 pc=0x1c59f5a
2023-02-14T02:07:42.628390492+08:00 stderr F go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x3adbcd8, 0xc00069eed0}, {0x35ff8e0, 0xc00028f400}, 0xc000492560, 0xc000492580)
2023-02-14T02:07:42.628407169+08:00 stderr F    /go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.37.0/interceptor.go:349 +0x528 fp=0xc000f83938 sp=0xc000f835c8 pc=0x19784c8
2023-02-14T02:07:42.628426255+08:00 stderr F github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3adbcd8?, 0xc00069eed0?}, {0x35ff8e0?, 0xc00028f400?})
2023-02-14T02:07:42.628439503+08:00 stderr F    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc000f83978 sp=0xc000f83938 pc=0x1c59f5a
2023-02-14T02:07:42.628458155+08:00 stderr F github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x3adbcd8, 0xc00069eed0}, {0x35ff8e0, 0xc00028f400}, 0xc001303a18?, 0x335b660?)
2023-02-14T02:07:42.62847807+08:00 stderr F     /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf fp=0xc000f839d0 sp=0xc000f83978 pc=0x1c59dff
2023-02-14T02:07:42.628493001+08:00 stderr F github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_LoadSegments_Handler({0x361a080?, 0xc000c9dcc0}, {0x3adbcd8, 0xc00069eed0}, 0xc000f7d3b0, 0xc00069e6f0)
2023-02-14T02:07:42.628506496+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5615 +0x138 fp=0xc000f83a28 sp=0xc000f839d0 pc=0x248fb38
2023-02-14T02:07:42.628529632+08:00 stderr F google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004a5a40, {0x3aec1c8, 0xc001003040}, 0xc001322000, 0xc00069e870, 0x4e6dc38, 0x0)
2023-02-14T02:07:42.628544504+08:00 stderr F    /go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13 fp=0xc000f83e48 sp=0xc000f83a28 pc=0x175ea93
2023-02-14T02:07:42.628564047+08:00 stderr F google.golang.org/grpc.(*Server).handleStream(0xc0004a5a40, {0x3aec1c8, 0xc001003040}, 0xc001322000, 0x0)
2023-02-14T02:07:42.628581019+08:00 stderr F    /go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b fp=0xc000f83f68 sp=0xc000f83e48 pc=0x1763a9b

```

**Comment by xiaofan-luan** [comment]
should be still the diskANN issue

**Comment by jiaoew1991** [comment]
the same as https://github.com/milvus-io/milvus/issues/22123

**Comment by NicoYuan1986** [comment]
Reproduced continuously.
link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/283/pipeline/135
log: [artifacts-milvus-distributed-kafka-nightly-283-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/283/artifact/artifacts-milvus-distributed-kafka-nightly-283-pymilvus-e2e-logs.tar.gz)

**Comment by NicoYuan1986** [comment]
Things don't seem to be getting better.
link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/284/pipeline/136
log: [artifacts-milvus-distributed-kafka-nightly-284-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/284/artifact/artifacts-milvus-distributed-kafka-nightly-284-pymilvus-e2e-logs.tar.gz)

**Comment by liliu-z** [comment]
> the same as #22123

No, not the same, #22133 is caused by aio issue which is get fixed. 

from the log 
```
[2023-02-15T19:25:09.677Z] [2023-02-15 19:24:41 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=failed to load collection, err=no enough nodes to create replicas[NoEnoughNode])>, <Time:{'RPC start': '2023-02-15 19:24:41.525926', 'RPC error': '2023-02-15 19:24:41.531232'}> (decorators.py:108)
```
This is caused by Milvus' side setting. 
/assign @xige-16

**Comment by NicoYuan1986** [comment]
Update: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/285/pipeline/155

1. milvus: b25b3ef
2. pymilvus: 2.3.0.dev35
3. log: [artifacts-milvus-distributed-kafka-nightly-285-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/285/artifact/artifacts-milvus-distributed-kafka-nightly-285-pymilvus-e2e-logs.tar.gz)
4. error message: 
```
[2023-02-16T19:26:50.527Z] [2023-02-16 19:26:04 - ERROR - ci_test]: Traceback (most recent call last):
[2023-02-16T19:26:50.527Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
[2023-02-16T19:26:50.527Z]     res = func(*args, **_kwargs)
[2023-02-16T19:26:50.527Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
[2023-02-16T19:26:50.527Z]     return func(*arg, **kwargs)
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 366, in load
[2023-02-16T19:26:50.527Z]     conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2023-02-16T19:26:50.527Z]     raise e
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2023-02-16T19:26:50.527Z]     return func(*args, **kwargs)
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2023-02-16T19:26:50.527Z]     ret = func(self, *args, **kwargs)
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
[2023-02-16T19:26:50.527Z]     raise e
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2023-02-16T19:26:50.527Z]     return func(self, *args, **kwargs)
[2023-02-16T19:26:50.527Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 661, in load_collection
[2023-02-16T19:26:50.527Z]     raise MilvusException(response.error_code, response.reason)
[2023-02-16T19:26:50.527Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to load collection, err=failed to spawn replica for collection[nodes not enough])>
[2023-02-16T19:26:50.527Z]  (api_request.py:39)
[2023-02-16T19:26:50.527Z] [2023-02-16 19:26:04 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=failed to load collection, err=failed to spawn replica for collection[nodes not enough])> (api_request.py:40)
```
5. milvus log: 
```
2023-02-17T03:05:25.930809363+08:00 stdout F [2023/02/16 19:05:25.930 +00:00] [DEBUG] [querynode/service.go:111] ["QueryNode connect to etcd successfully"]
2023-02-17T03:05:26.010330194+08:00 stderr F panic: runtime error: invalid memory address or nil pointer dereference
2023-02-17T03:05:26.010349834+08:00 stderr F [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x29d670b]
2023-02-17T03:05:26.010353073+08:00 stderr F
2023-02-17T03:05:26.010367309+08:00 stderr F goroutine 303 [running]:
2023-02-17T03:05:26.010370237+08:00 stderr F panic({0x335ddc0, 0x4e5bf00})
2023-02-17T03:05:26.010373406+08:00 stderr F    /usr/local/go/src/runtime/panic.go:941 +0x397 fp=0xc000646ff8 sp=0xc000646f38 pc=0x11edcf7
2023-02-17T03:05:26.010375744+08:00 stderr F runtime.panicmem(...)
2023-02-17T03:05:26.010378373+08:00 stderr F    /usr/local/go/src/runtime/panic.go:220
2023-02-17T03:05:26.010380698+08:00 stderr F runtime.sigpanic()
2023-02-17T03:05:26.010394771+08:00 stderr F    /usr/local/go/src/runtime/signal_unix.go:818 +0x336 fp=0xc000647048 sp=0xc000646ff8 pc=0x1204d16
2023-02-17T03:05:26.010397702+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*QueryNode).LoadSegments(0xc001093c00, {0x3add658?, 0xc000eb1bc0}, 0xc000dc39a0)
2023-02-17T03:05:26.01040035+08:00 stderr F     /go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:432 +0x8b fp=0xc0006474e8 sp=0xc000647048 pc=0x29d670b
2023-02-17T03:05:26.010403116+08:00 stderr F github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).LoadSegments(0x0?, {0x3add658?, 0xc000eb1bc0?}, 0x0?)
2023-02-17T03:05:26.010414575+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:270 +0x2b fp=0xc000647518 sp=0xc0006474e8 pc=0x2bbeb4b
2023-02-17T03:05:26.010428984+08:00 stderr F github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_LoadSegments_Handler.func1({0x3add658, 0xc000eb1bc0}, {0x3600c80?, 0xc000dc39a0})
2023-02-17T03:05:26.01043333+08:00 stderr F     /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5613 +0x78 fp=0xc000647558 sp=0xc000647518 pc=0x2490338
2023-02-17T03:05:26.010443811+08:00 stderr F github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor({0x3add658?, 0xc000eb1bc0?}, {0x3600c80, 0xc000dc39a0}, 0xc0006475b8?, 0xc000f442d0)
2023-02-17T03:05:26.010457658+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22 +0x49 fp=0xc000647588 sp=0xc000647558 pc=0x2475da9
2023-02-17T03:05:26.010468534+08:00 stderr F github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3add658?, 0xc000eb1bc0?}, {0x3600c80?, 0xc000dc39a0?})
2023-02-17T03:05:26.010475256+08:00 stderr F    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc0006475c8 sp=0xc000647588 pc=0x1c5a23a
2023-02-17T03:05:26.010488598+08:00 stderr F go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x3add658, 0xc000f89740}, {0x3600c80, 0xc000dc39a0}, 0xc00107b300, 0xc00107b320)
```

**Comment by NicoYuan1986** [comment]
The same errors happen every day with hundreds of cases failed. I think we should solve it as soon as possible.

**Comment by smellthemoon** [comment]
seems similar to #22289 , fixed with  #22290. you can check it after the pr merged. @NicoYuan1986

**Comment by NicoYuan1986** [comment]
Cheers! Fixed!
